test/sagemaker_tests/huggingface_pytorch/training/integration/sagemaker/test_trcomp.py

```diff
@@ -107,106 +136,280 @@ def test_trcomp_default(self, patched, ecr_image, sagemaker_session, tmpdir, py_
                 py_version=py_version,
                 max_retry_attempts=15,
             )
-            estimator.fit(job_name=sagemaker.utils.unique_name_from_base('hf-pt-trcomp-single-gpu-default'), logs=True)
+            estimator.fit(
+                job_name=sagemaker.utils.unique_name_from_base("hf-pt-trcomp-SNSG-default"),
+                logs=True,
+            )
         captured = capsys.readouterr()
-        logs = captured.out+captured.err
+        logs = captured.out + captured.err
         assert "Found configuration for Training Compiler" in logs
         assert "Configuring SM Training Compiler" in logs
         assert "device: xla" in logs
 
-
     @pytest.mark.model("bert-large")
-    def test_trcomp_enabled(self, patched, ecr_image, sagemaker_session, tmpdir, py_version, capsys):
-        '''
+    def test_trcomp_enabled(
+        self,
+        patched,
+        ecr_image,
+        sagemaker_session,
+        tmpdir,
+        py_version,
+        capsys,
+        instance_type,
+        instance_count,
+    ):
+        """
         Tests the explicit enabled configuration of SM trcomp
-        '''
+        """
         transformers_version = get_transformers_version(ecr_image)
-        git_config = {'repo': 'https://github.com/huggingface/transformers.git', 'branch': 'v'+transformers_version}
+        git_config = {
+            "repo": "https://github.com/huggingface/transformers.git",
+            "branch": "v" + transformers_version,
+        }
 
-        instance_count = 1
-        instance_type = "ml.p3.2xlarge"
-        
         source_dir = (
             "./examples/question-answering"
             if Version(transformers_version) < Version("4.6")
             else "./examples/pytorch/question-answering"
         )
-        
+
         with timeout(minutes=DEFAULT_TIMEOUT):
             estimator = HuggingFace(
                 compiler_config=TrainingCompilerConfig(enabled=True),
-                entry_point='run_qa.py',
+                entry_point="run_qa.py",
                 source_dir=source_dir,
                 git_config=git_config,
                 metric_definitions=metric_definitions,
-                role='SageMakerRole',
+                role="SageMakerRole",
                 image_uri=ecr_image,
                 instance_count=instance_count,
                 instance_type=instance_type,
                 sagemaker_session=sagemaker_session,
                 hyperparameters=hyperparameters,
-                environment={'GPU_NUM_DEVICES':'1'},
                 py_version=py_version,
                 max_retry_attempts=15,
             )
-            estimator.fit(job_name=sagemaker.utils.unique_name_from_base('hf-pt-trcomp-single-gpu-enabled'), logs=True)
+            estimator.fit(
+                job_name=sagemaker.utils.unique_name_from_base("hf-pt-trcomp-SNSG-enabled"),
```

Comment:
nitpick: please expand SNSG to day single-node-single-gpu. It's not obvious.

Reply:
Tried that, it gets truncated because the name is too long